src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -729,3 +776,76 @@ object OutputType extends FgBioEnum[OutputType] {
   case object Bam extends OutputType { val producesBam: Boolean = true; val producesFastq: Boolean = false; }
   case object BamAndFastq extends OutputType { val producesBam: Boolean = true; val producesFastq: Boolean = true; }
 }
+
+/** A little class to store read-level information from the comment in a FASTQ read name that is in Illumina
+  * standard format:
+  *
+  * `<read>:<is filtered>:<control number>:<barcode-sequence>`
+  *
+  * This is typically inferred from the first comment in the input FASTQ header.
+  *
+  * @param readNumber the read number
+  * @param passQc true if the read passes quality filters, false if it fails
+  * @param internalControl true if the read is an internal control, false otherwise
+  * @param sampleInfo additional sample information, such as sample barcode (bcl2fastq/BCL convert) or sample number (BaseSpace)
+  * @param rest any additional information beyond the comment.
+  */
+case class ReadInfo(readNumber: Int, passQc: Boolean, internalControl: Boolean, sampleInfo: String, rest: Seq[String]) {
+  override def toString: String = {
+    val leading = f"$readNumber:${if (passQc) "Y" else "N"}:${if (internalControl) 1 else 0}:$sampleInfo"
+    if (rest.isEmpty) leading else leading + " " + rest.mkString(" ")
+  }
+}
+
+object ReadInfo {
+  /** Builds the [[ReadInfo]] by parsing a [[FastqRecord]]. */
+  def apply(rec: FastqRecord): ReadInfo = this(rec.name, rec.comment)
+
+  /** Builds the [[ReadInfo]] by parsing a [[DemuxRecord]]. */
+  def apply(rec: DemuxRecord): ReadInfo = this(rec.name, rec.comment)
+
+  /** Builds the [[ReadInfo]] by parsing a standard input FASTQ. */
+  def apply(name: String, comment: Option[String]): ReadInfo = try {
+    val commentFields = comment.getOrElse {
+      throw new IllegalArgumentException(s"Comment not found.")
+    }.split(' ')
```

Comment:
If the comment is required, the type should not be `Option[String]` but just `String`.

Reply:
I also don't really understand what's going on here.  Do people really generate fastqs in the wild with multiple space-separated comments?  If so I think it would make more sense to just call this `comments` since we then break down the first comment into _it's_ constituent fields.

Reply:
Yes, there are space separated comments in the wild.  I like you're suggestion too.